config/environments/production.rb

```diff
@@ -128,8 +128,8 @@
     config.action_mailer.raise_delivery_errors = Settings.mail.raise_delivery_errors
 
     config.action_mailer.default_url_options = {
-      host: Settings.host_domain,
-      protocol: Settings.host_protocol
+      host: ENV["TLS_DOMAIN"] || Settings.host_domain,
+      protocol: (ENV["TLS_DOMAIN"].present? || Settings.host_protocol) ? "https" : "http"
```

Comment:
I'm not sure about Settings.host_protocol and Settings.host_domain. Do we need to mix them with `TLS_DOMAIN`? Or, shouldn't we touch existing settings? Or, do we need to remove these two settings completely? Each approach have different disadvantages.